lib/mix/lib/mix/scm/git.ex

```diff
@@ -7,7 +7,14 @@ defmodule Mix.SCM.Git do
   end
 
   def format(opts) do
-    opts[:git]
+    {ref_type, ref_value} = cond do
+      opts[:tag] -> {:tag, opts[:tag]}
+      opts[:branch] -> {:branch, opts[:branch]}
+      opts[:ref] -> {:ref, opts[:ref]}
+      true -> {nil, nil}
+    end
+    extra_info = if ref_type, do: " - #{ref_type}: #{ref_value}", else: ""
+    "#{opts[:git]}#{extra_info}"
```

Comment:
There is already some code related to this here: https://github.com/elixir-lang/elixir/pull/10147/files#diff-dac65d3a4f087adb3405c3e1e64fd076R25

Maybe we can refactor and reuse it. Something like:

    "#{opts[:git]}#{extra_info(opts)}"

Reply:
Also please take a look at the tests. :)

Reply:
will take a look, thx

Reply:
think you posted a link to this one :)

Reply:
@josevalim

Reply:
Oops, you have to click the expand button below to see it. Here is a better link: https://github.com/elixir-lang/elixir/blob/master/lib/mix/lib/mix/scm/git.ex#L18-L22